src/BloomExe/Edit/JpegWarningDialog.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using L10NSharp;
+
+namespace Bloom.Edit
+{
+	public partial class JpegWarningDialog : Form
+	{
+		public JpegWarningDialog()
+		{
+			InitializeComponent();
+			_warningText.Text = LocalizationManager.GetString("EditTab.JpegWarningDialog.WarningText",
+				@"The file you’ve chosen is a “jpeg” file. JPEGs are perfect for photographs and color artwork. However, JPEGs are a big problem for black and white line art. Problems include:
+• Fuzziness and grey dots.
+• Large file sizes, making the book hard to share.
+• If there are many large JPEGs, Bloom may not have enough memory to make PDFs.
+
+Note: Because JPEG is “lossy”, converting a JPEG to PNG, TIFF, or bmp actually makes things even worse. If this is black and white line art, you want to get an original scan in one of those formats.
+
+Please select from one of the following, then click “OK”:").Replace("\r\n","\n").Replace("\n", Environment.NewLine);
+		}
+
+		private void _okButton_Click(object sender, EventArgs e)
+		{
+			DialogResult = _cancelRadioButton.Checked ? DialogResult.Cancel : DialogResult.OK;
+			Close();
+		}
+
+		/// <summary>
+		/// Tells whether the supplied image looks suspiciously like a file that 
+		/// would be better off as a PNG. Not particularly smart, but fast.
+		/// </summary>
+		public static Boolean ShouldWarnAboutJpeg(Image image)
+		{
+			var bmp = image as Bitmap;
+			if (bmp == null)
+				return false;
+			return HasLotsOfWhiteSpace(bmp) && (IsGreyScale(bmp) || !HasLotsOfColor(bmp));
```

Comment:
I see why something is likely to be line art if it has lots of white space and few colors, but why is something grey likely to be line art even if it has many shades of gray? Or do we WANT to warn for (say) a B/W photo with the background clipped to white? Of course you're only approximating and it's not a disaster to give this message when it's not needed...but a comment might help.